### PR TITLE
Tiny API review renames

### PR DESCRIPTION
**Pull request by roji**
Part of #30306

## Stuff I did:

> @roji Is there a better name for SqlNullabilityProcessor.PreferExistsToComplexIn? What makes it complex?

Renamed it to PreferExistsToInWithCoalesce, which explains better when this is used (but still sucks).

>  TranslateCollection should be renamed to TranslateScalarCollection

I ended up renaming to TranslatePrimitiveCollection since that's the naming we ended up using e.g. in the user-facing builder API.

## Stuff I left as-is for now:

> @roji Is there a better name for InlineQueryRootExpression? It seems too generic. Maybe something with *Collection*

I thought about it some more; a query root by definition represents some sort of collection, so I really think adding `Collection` is redundant (note that we'd also have to do it for `ParameterQueryRootExpression`)... For comparison, existing type names for query roots are `EntityQueryRootExpression`, `SqlQueryRootExpression` and `TableValuedFunctionQueryRootExpression`; I think `InlineQueryRootExpression` and `ParameterQueryRootExpression` fit well into this and are reasonably descriptive - but let me know if there are strong feelings here and I will change.

> QueryRootProcessor.ShouldConvertToInlineQueryRoot too

As above.

> @roji Is there a better name for QuerySqlGenerator.TryGenerateWithoutWrappingSelect? It doesn't follow the typical Try- pattern.

Looking at it, this does return a bool to indicate success/failure - though it indeed has no out var. I *think* that's OK and in-line with established patterns - [`DateTime.TryFormat`](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tryformat?view=net-7.0) is a good example of an API which behaves just like this. The rest of the name is a bit long/wonky, but I can't really come up with some better... In some cases we have a `SelectExpression` wrapping something where the `SelectExpression` doesn't actually need to be generated in SQL (one example is `X UNION y`, which we represent as wrapped by a `SelectExpression` although it isn't outputted). This method is what identifies these cases and also writes the SQL out; we could also shorten this to `TryGenerateUnwrapped`, which is shorter but less descriptive - am open to other ideas.

/cc @danmoseley

### CODE REVIEW

**Review by ajcvickers — APPROVED** [review]

**Comment by danmoseley** [comment]
thanks, after EOD today it's tactics mode and this would be an RC2 thing I think.

**Comment by ajcvickers** [comment]
@danmoseley No more PRs planned for RC1 from @roji or myself.
